--- a/src/select/pselect.c
+++ b/src/select/pselect.c
@@ -9,7 +9,7 @@
 
 int pselect(int n, fd_set *restrict rfds, fd_set *restrict wfds, fd_set *restrict efds, const struct timespec *restrict ts, const sigset_t *restrict mask)
 {
-	syscall_arg_t data[2] = { (uintptr_t)mask, _NSIG/8 };
+	syscall_arg_t data[2] = { (syscall_arg_t)mask, (syscall_arg_t)(_NSIG/8) };
 	time_t s = ts ? ts->tv_sec : 0;
 	long ns = ts ? ts->tv_nsec : 0;
 #ifdef SYS_pselect6_time64
